osm2gtfs/core/osm_connector.py

```diff
@@ -345,57 +355,134 @@ def _build_route_variant(self, route_variant, query_result_set, rm=None):
-        if 'name' not in stop.tags:
-            stop.tags['name'] = "[" + self.stop_no_name + "]"
+        if self._is_valid_stop_candidate(stop):
+
+            # Make sure name is not empty
+            if 'name' not in stop.tags:
+                stop.tags['name'] = "[" + self.stop_no_name + "]"
+
+            # Ways don't have a pair of coordinates and need to be calculated
+            if osm_type == "way":
+                (stop.lat, stop.lon) = Helper.get_center_of_nodes(
+                    stop.get_nodes())
+
+            # Create and return Stop object
+            stop = Stop(osm_id=stop.id, osm_type=osm_type, tags=stop.tags,
+                        lat=stop.lat, lon=stop.lon, name=stop.tags['name'])
+            return stop
+
+        else:
+            sys.stderr.write(
+                "Warning: Potential stop was not approved and is ignored")
+            sys.stderr.write(
+                " Check tagging: http://osm.org/" + osm_type + "/" + str(
+                    stop.id) + "\n")
+            return False
+
+    def _build_station(self, stop_area, osm_type):
+        """Helper function to build Station objects from stop_areas
 
-        # Ways don't have coordinates and they have to be calculated
-        if stop_type == "way":
-            (stop.lat, stop.lon) = Stop.get_center_of_nodes(stop.get_nodes())
+        The function creates a Station object for the stop_area
+        flagged as location_type = 1. This means station, that can
+        group stops.
 
-        s = Stop(stop.id, "node", stop.tags['name'], stop.lat, stop.lon)
-        return s
+        The members of this relation add this station their parent.
 
-    def _build_stop_area(self, relation):
-        """Helper function to build a StopArea object
+        Returns a initiated Station object from raw data
 
-        Returns a initiated StopArea object from raw data
         """
-        stop_members = {}
-        for member in relation.members:
-            if (isinstance(member, overpy.RelationNode) and
-               member.role == "platform"):
-                stop = self.stops.pop("node/" + str(member.ref))
-                stop_members["node/" + str(member.ref)] = stop
-
-        if 'name' not in relation.tags:
-            sys.stderr.write("Stop area without name." +
-                             " Please fix in OpenStreetMap\n")
-            sys.stderr.write("http://osm.org/relation/" +
-                             str(relation.id) + "\n")
-            stop_area = StopArea(relation.id, stop_members, self.stop_no_name)
+
+        # Check whether a valid stop_area candidade
+        if 'public_transport' in stop_area.tags and stop_area.tags[
+                        'public_transport'] == 'stop_area':
+
+            # Analzyse member objects (stops) of this stop area
+            members = {}
+            for member in stop_area.members:
+                if (isinstance(member, overpy.RelationNode) and
+                   member.role == "platform"):
```

Comment:
Shouldn't we also accept `stop_position`? And shouldn't we also accept `platform`s as `way`s?

Reply:
Yes we should! But this is how it is implemented at the moment (in master), and I would rather change this in a subsequent step (separate PR).

Reply:
Yes, please no unrelated changes. This PR is too big as it is already. Changes in existing functionality should be made in separate PRs.

Reply:
No worry, see #100.